docs/how-to/hip_runtime_api/memory_management/host_memory.rst

```diff
@@ -0,0 +1,205 @@
+operations, such as paging, which can negatively impact the overall performance
+of the host.
+
+The example code how to use pinned memory in HIP showed at the following example.
+
+.. code-block:: cpp
+
+  #define ELEMENT_NUMBER 100
+
+  int *host_input, *host_output;
+  // Host allocation
+  hipHostMalloc((int **)&host_input, ELEMENT_NUMBER * sizeof(int));
+  hipHostMalloc((int **)&host_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Host data preparation
+  for (int i = 0; i < ELEMENT_NUMBER; i++) {
+    host_input[i] = i;
+  }
+  memset(host_output, 0, ELEMENT_NUMBER * sizeof(int));
+
+  int *device_input, *device_output;
+
+  // Device allocation
+  hipMalloc((int **)&device_input, ELEMENT_NUMBER * sizeof(int));
+  hipMalloc((int **)&device_output, ELEMENT_NUMBER * sizeof(int));
+
+  // Device data preparation
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyHostToDevice);
+  hipMemset()
+
+  // Run the kernel
+  ...
+
+  hipMemcpy(device_input, host_input, ELEMENT_NUMBER * sizeof(int), cudaMemcpyDeviceToHost);
+
+  // Free host memory
+  delete[] host_input;
+  delete[] host_output;
+
+  // Free device memory
+  hipFree()
+  hipFree()
+
+The pinned memory allocation is effected with different flags, which details
+described at :ref:`memory_allocation_flags`.
+
+.. _memory_allocation_flags:
+
+Memory allocation flags of pinned memory
+--------------------------------------------------------------------------------
+
+The ``hipHostMalloc`` flags specify different memory allocation types for pinned
+host memory:
+
+* ``hipHostMallocPortable``: The memory is considered allocated by all contexts,
+  not just the one on which the allocation is made.
+* ``hipHostMallocMapped``: Map the allocation into the address space for
+  the current device, and the device pointer can be obtained with
+  :cpp:func:`hipHostGetDevicePointer`.
+* ``hipHostMallocNumaUser``: The flag to allow host memory allocation to
+  follow Numa policy by user. Target of Numa policy is to select a CPU that is
+  closest to each GPU. Numa distance is the measurement of how far between GPU
+  and CPU devices.
+* ``hipHostMallocWriteCombined``: Allocates the memory as write-combined. On
+  some system configurations, write-combined allocation may be transferred
+  faster across the PCI Express bus, however, could have low read efficiency by
+  most CPUs. It's a good option for data transfer from host to device via mapped
+  pinned memory.
+* ``hipHostMallocCoherent``: Allocate fine-grained memory. Overrides
+  ``HIP_HOST_COHERENT`` environment variable for specific allocation. For
+  further details, check :ref:`coherence_control`.
+* ``hipHostMallocNonCoherent``: Allocate coarse-grained memory. Overrides
+  ``HIP_HOST_COHERENT`` environment variable for specific allocation. For
+  further details, check :ref:`coherence_control`.
+
+All allocation flags are independent and can be used in most of the combination
```

Comment:
"most combinations" would mean, that they are not independent

Reply:
Fixed.